fedbiomed/researcher/secagg/_secure_aggregation.py

```diff
@@ -331,18 +334,18 @@ def aggregate(
     def _validate(
         self,
         aggregate: functools.partial,
-        encryption_factors: List[List[int]],
-        num_expected_params: int | None = None
+        encryption_factors: Dict[str, Union[List[int], None]],
+        num_expected_params: int | None = None,
     ) -> bool:
         """Validate given inputs"""
 
-
-        if encryption_factors is None:
+        # at this point we know `isinstance(encryption_factors, dict) and len(encryption_factors) == num_nodes`
+        if any([v is None for v in encryption_factors.values()]):
```

Comment:
if think `all(encryption_factors.values())` does the same think. 

```suggestion
        if not all(encryption_factors.values()):
```

Reply:
This is not exactly the same test. For example with `encryption_factors.values = [False, [1]]` tests don't return the same value. Testing specifically for `None` value intends to catch only the case of setup mismatch between researcher and node, while doing a `not all()` test will also catch some malicious attack cases, misleading the user to believe this is a setup mismatch (while malicious case are caught in `SecaggCrypter` with explicit message indicating the type of the encryption factors is not correct).

Sio I would prefer to keep the current more explicit test